src/dos/dos_devices.cpp

```diff
@@ -375,6 +383,61 @@ DOS_File &DOS_File::operator=(const DOS_File &orig)
 	return *this;
 }
 
+void DOS_SetMediaDelay(const bool use_media_delay)
+{
+	auto get_delay_percent = []() {
+		LOG_MSG("machine = %d", machine);
+		switch (machine) {
+		case MCH_HERC: return 300;
+		case MCH_CGA:
+		case MCH_TANDY:
+		case MCH_PCJR: return 150;
+		case MCH_EGA: return 125;
+		case MCH_VGA:
+		default: return 100;
+		}
+	};
+
+	DOS_File::MediaDelayGuard::SetDelayPercent(
+	        use_media_delay ? get_delay_percent() : 0);
+
+	LOG_MSG("DOS: Media will %s delays",
+	        use_media_delay ? "use realistic" : "run without");
+}
+
+DOS_File::MediaDelayGuard::~MediaDelayGuard()
+{
+	if (delay_percentage == 0) {
+		return;
+	}
+	auto scaled_ops = (GetNumOperations() * delay_percentage) / 100;
+	while (scaled_ops--) {
+		CALLBACK_Idle();
+	}
+}
+
+constexpr void DOS_File::MediaDelayGuard::SetDelayPercent(const int percentage) noexcept
+{
+	assert(percentage >= 0);
+	delay_percentage = percentage;
+}
+
+constexpr uint16_t DOS_File::MediaDelayGuard::GetNumOperations() const noexcept
+{
+	constexpr uint16_t ReadBytesPerOp  = 128;
+	constexpr uint16_t WriteBytesPerOp = 192;
+	constexpr uint16_t SeekMultiplier  = 20;
+
+	// clang-format off
+
+	return (bytes_read / ReadBytesPerOp)
+	     + (bytes_written / WriteBytesPerOp)
+	     + (num_seeks * SeekMultiplier)
+	     + num_closes;
```

Comment:
So this is the problem I was trying to describe with the Seek delay.  You're adding a delay when a game seeks within the same file (presumably to account for the drive head moving) but you have no such delay (that I can tell) when a game reads from one file and then immediately afterwards reads from a different file which could be on the other side of the drive.

Seeking within the same file shouldn't cause much of a delay in the first place.  Barring fragmentation, the bytes should be contiguous.  The drive head delay really comes in when you're reading from multiple files spread across the disk.

If we really wanted to be accurate we would need to keep track of what the last file we read or wrote to is.  I wasn't suggesting that for this PR though.  I was just suggesting to remove the seek delay entirely as, IMO, it doesn't make sense the way it's implemented here.  Let me know if I'm missing something though.